### DISCUSSION TITLE
'git add' - not working!

### DISCUSSION THREAD

**Discussion by echocage123** [community/community#29288] [Other Feature Feedback, Questions, & Ideas]
"git add" - command successing without troubles, but after executing "git commit -m 'some changes' " - its throwing "nothing added to commit but untracked files present (use "git add" to track)
"

**Comment by airtower-luna** [comment]
Sounds a bit like maybe there wasn't any change in the file(s) you added, but it's hard to say for sure without more details. What would help:

* The exact `git add` command you used
* `git status` output before and after `git add`

**Reply by echocage123** [reply]
I can't not have! After all, I just created a repository and I'm trying to send 1 file and 1 folder with 1 file to it.

Похоже, что в добавленных вами файлах не было никаких изменений, но трудно сказать наверняка без более подробной информации. Что поможет:

Точная git addкоманда, которую вы использовали
git statusвывод до и послеgit add
I can't not have! After all, I just created a repository and I'm trying to send 1 file and 1 folder with 1 file to it.

What requests I made:
git add include/
git add srvprint_lib.sp
git commit -m 'Release!'

And after 'git commit', im getting
`$ git commit -m 'Release!'
warning: could not open directory 'Application Data/': Permission denied
warning: could not open directory 'Cookies/': Permission denied
warning: could not open directory 'Intel/Logs/': Permission denied
warning: could not open directory 'Local Settings/': Permission denied
warning: could not open directory 'NetHood/': Permission denied
warning: could not open directory 'PrintHood/': Permission denied
warning: could not open directory 'Recent/': Permission denied
warning: could not open directory 'SendTo/': Permission denied
warning: could not open directory 'главное меню/': Permission denied
warning: could not open directory 'Мои документы/': Permission denied
warning: could not open directory 'Шаблоны/': Permission denied
On branch main
Untracked files:
(use "git add ..." to include in what will be committed)
../../../.DevelNext/
../../../.MS.dat
../../../.VirtualBox/
../../../.atom/
../../../.bash_history
../../../.dbvis/
../../../.dotnet/
../../../.gitconfig
../../../.gitignore
../../../.htaccess
../../../.idea/
../../../.idlerc/
../../../.keras/
../../../.lesshst
../../../.matplotlib/
../../../.node_repl_history
../../../.nuget/
../../../.oracle_jre_usage/
../../../.packettracer
../../../.ssh/
../../../.viminfo
../../../.vscode/
../../../123/
../../../2BF4D6EC09543A398F.data
../../../3D Objects/
../../../AppData/
../../../Cisco Packet Tracer 6.2sv/
../../../Contacts/
../../../Creative Cloud Files/
../../../Csgo_Weapon_healthshot.png
../../../DarkMatter/
../../.vs/
../../008.png
../../1.svg
../../2x2maps.txt
../../Adobe Photoshop 2021.lnk
../../Adobe Premiere Pro 2020.lnk
../../Arma 3.url
../../Arma Reforger.lnk
../../Atom.lnk
../../Authy Desktop.lnk
../../BeamNG drive.lnk
../../BeamNG.drive.lnk
../../Cheat Engine.lnk
../../Cisco Packet Tracer Student.lnk
../../Counter-Strike Global Offensive - SDK.url
../../Counter-Strike Global Offensive.url
../../Discord.lnk
../../ExLoader.lnk
../../Font Manager.lnk
../../Furios.lnk
../../GTFO.url
../../GitHub Desktop.lnk
../../Grand Theft Auto V.url
../../Melonity.lnk
../../Minority.lnk
../../New Text Document.txt
../../New folder/
../../New_Plugin_13 (1).sp
../../OSPF.pkt
../../OpenIV.lnk
../../PLUGIN_backup1454/
../../PUBG LITE.lnk
../../Postal 2 Complete.lnk
../../Process Hacker 2.lnk
../../REKLAMAFURIOS/
../../Rockstar Games Launcher.lnk
"../../Rust (\320\261\320\265\320\267 \320\273\320\260\321\203\320\275\321\207\320\265\321\200\320\260).lnk"
../../Rust.lnk
../../SPCode.lnk
"../../SQL \321\210\320\277\320\260\321\200\320\263\320\260\320\273\320\272\320\260.txt"
../../Sandboxed Web Browser.lnk
../../Screenshot_1.png
../../Screenshot_2.png
../../Screenshot_3.png
../../Screenshot_4.png
../../Screenshot_5.png
../../Screenshot_6.png
../../Screenshot_7.png
../../Screenshot_8.png
../../SpawnSystem.sp
../../Spotify.lnk
../../Telegram.lnk
../../Termius.lnk
../../Unreal Engine.lnk
../../VPN PRO.lnk
../../Visual Studio Code.lnk
../../YouTube.lnk
../../aboba_maps/
../../ae.smx
../../ae.sp
../../animoverlay/
../../blender.lnk
../../csgoserver-2.cfg
../../csgoserver-3.cfg
../../csgoserver-4.cfg
../../csgoserver.cfg
../../dataPro.lnk
../../deathmatch.sp
../../desktop.ini
../../furios_ranks/
../../git/
../../gnida/
../../hammerplusplus/
../../i8080emu.lnk
../../laAST.pkt
../../learn/
../../main.pkt
../../map-decals.smx
../../map-decals.sp
../../maxmoney.sp
../../nixware (1).exe
../../nixware.log
../../office.pkt
../../oqest.txt
../../overlays/
../../passs/
../../php.txt
../2x2_pro/
../MIRAGE/
../Material_admin/
../ParanoiaIPTracker.smx
../PluginBlock_Stuff.cfg
../afk_manager/
../anti-afk/
../anti-untrusted/
../antidefensiveprediction/
../autohealer/
../consolelog/
../databases.cfg
../directorydownloader/
../dmdm/
../eblo.smx
../eblo.sp
../fff/
../forcedownload/
../highstats_alert/
../map/
../mapcount/
../musor/
../my_reklama/
../noblood/
../nofd.smx
../overlay/
../pidor.smx
../pidor.sp
../pug/
../pugPlugs/
../pugsetup.phrases.txt
../quest/
../regexfilter/
../restrictroll/
../rtv/
../savechat/
advertisement_translate.ini
csgo_colors.inc
hvh_test.sp
plugin.reklama_hvh.cfg
reklama_translate.sp
../umc/
../../primordial_loader.exe
../../public/
../../quests/
../../qwqwqwqwqw/
../../site.txt
../../sm_regexfilter.sp
../../sp compiler/
../../ssh/
../../usenominate.smx
../../work/
../../workNetwork.pkt
../../xampp/
../../ycrkgmmdmnfydqf.exe
"../../\302\265Torrent.lnk"
"../../\320\220\321\200\320\265\320\275\320\264\320\260 \321\201\320\265\321\200\320\262\320\265\321\200\320\276\320\262 - \321\204\320\270\320\275\320\260\320\275\321\201\321\213.txt"
"../../\320\220\321\200\321\205\320\270\320\262 WinRAR.rar"
"../../\320\267\320\277.drawio"
"../../\320\274\320\260\321\201\320\272\320\260 \321\201\320\265\321\202\320\270.drawio"
../../../DevelNextProjects/
../../../Documents/
../../../Downloads/
../../../Favorites/
../../../Links/
../../../Local Disk (C) - Shortcut.lnk
../../../Manager.php
../../../MegaSAS.log
../../../MicrosoftEdgeBackups/
../../../Midnight Commander/
../../../Music/
../../../My Games/
../../../NTUSER.DAT
../../../NTUSER.DAT{53b39e88-18c4-11ea-a811-000d3aa4692b}.TM.blf
../../../NTUSER.DAT{53b39e88-18c4-11ea-a811-000d3aa4692b}.TMContainer00000000000000000001.regtrans-ms
../../../NTUSER.DAT{53b39e88-18c4-11ea-a811-000d3aa4692b}.TMContainer00000000000000000002.regtrans-ms
../../../OneDrive/
../../../PhpstormProjects/
../../../Pictures/
../../../PycharmProjects/
../../../Saved Games/
../../../Searches/
../../../VIP_HealthshotPro.cfg
../../../VIP_Kill_Bonus.cfg
../../../Videos/
../../../VirtualBox VMs/
../../../YandexDisk/
../../../admin_levels.cfg
../../../admin_overrides.cfg
../../../adminmenu_cfgs.txt
../../../app.js
../../../blacklist.plugins.txt
../../../common.phrases.txt
../../../composer-setup.php
../../../composer.json
../../../composer.lock
../../../composer.phar
../../../connecting.php
../../../create_table.sql
../../../cs_italy.cfg
../../../csgo_gunmenu 1.2.smx
../../../csgoserver.cfg
../../../deathmatch.ini
../../../downloadlist.txt
../../../errors_20220707.log
../../../file.txt
../../../furiosl.png
../../../furiostransition.vmt
../../../groups.ini
../../../index.html
../../../index.php
../../../interface.php
../../../ip
../../../lr_base.sq3
../../../metamod.vdf
../../../multi1v1.cfg
../../../new/
../../../ntuser.dat.LOG1
../../../ntuser.dat.LOG2
../../../ntuser.ini
../../../office_arena_furios_2022.bsp
../../../package-lock.json
../../../package.json
../../../py
../../../readme.html
../../../resdfs.txt00
../../../resdfs.txt_01
../../../resdfs.txt_02.04
../../../skillgroup51.svg
../../../tblproduct.sql
../../../trails.cfg
../../../trails.ini
../../../training1.nav
../../../vendor/
nothing added to commit but untracked files present (use "git add" to track)
`

**Reply by airtower-luna** [reply]
It looks like you turned *you whole user directory* into a repository, instead of only the directory with your code. That will cause all kinds of trouble.

Most likely the best option is to delete that repository (not all your data!) and start fresh. Note that this will erase any history you already have. :warning: But if you just created the repository that should be fine, you can check with `git log`.

Assuming there really aren't any commits in the repository:

* Find the (hidden) `.git` directory and delete it. It's always at the top level of the repository.
* `cd` into the directory that you want to be your repository. The one with your source files.
* Run `git init`.
* Then try to add and commit again.

Also I recommend you check the [Git Book website](https://git-scm.com/book/en/v2). Look on the left for translations.

**Reply by echocage123** [reply]
I dont have .git folder now, but some earlier i has it - but because i had this problem, i removed this folder
![image](https://user-images.githubusercontent.com/67440446/184412940-0c9929ef-f73e-4b4e-b14f-6bbf10c72c20.png)

git log:
$ git log
commit a0cd8343c2689ef76e075961e70707760840ee7b (HEAD -> main)
Author: echocage123 <germankorsun@yandex.ru>
Date:   Fri Aug 12 18:44:46 2022 +0300

    Release!


-+-+-+-
+-+-+-+

After your instructions, I actually managed to make a commit, but now I get an error when git push:

`user@DESKTOP-3QAIDKC MINGW64 ~/Desktop/plugins/ultimatereklama (master)
$ git add srvprint_lib.sp

user@DESKTOP-3QAIDKC MINGW64 ~/Desktop/plugins/ultimatereklama (master)
$ git add include/

user@DESKTOP-3QAIDKC MINGW64 ~/Desktop/plugins/ultimatereklama (master)
$ git commit -m 'Release!'
[master (root-commit) ca42a79] Release!
 2 files changed, 99 insertions(+)
 create mode 100644 include/srvforcechat.inc
 create mode 100644 srvprint_lib.sp

user@DESKTOP-3QAIDKC MINGW64 ~/Desktop/plugins/ultimatereklama (master)
$ git push -u origin main
error: src refspec main does not match any
error: failed to push some refs to 'origin'

user@DESKTOP-3QAIDKC MINGW64 ~/Desktop/plugins/ultimatereklama (master)
$ git push -u main main
error: src refspec main does not match any
error: failed to push some refs to 'main'

user@DESKTOP-3QAIDKC MINGW64 ~/Desktop/plugins/ultimatereklama (master)
$ git push -u master main
error: src refspec main does not match any
error: failed to push some refs to 'master'

user@DESKTOP-3QAIDKC MINGW64 ~/Desktop/plugins/ultimatereklama (master)
$ git branch -M main

user@DESKTOP-3QAIDKC MINGW64 ~/Desktop/plugins/ultimatereklama (main)
$ git push -u master main
fatal: 'master' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
`

**Reply by airtower-luna** [reply]
`git push -u origin main` tries to push the branch `main` to the remote repository `origin`. At first that failed because your branch was called `master`. Then you renamed the branch to `main` with `git branch -M main`. Good.

The remaining problem: You're now trying to push to a remote called `master`, which doesn't exist.

If you have already added your GitHub repository as `origin`, you need to use:

```
git push -u origin main
```

See also: https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes

**Reply by AlexHo35** [reply]
Try : 
git add "file name"
